### PR TITLE
add more drivers

### PR DESCRIPTION
**Pull request by ocefpaf**
I skipped [`libkml`](https://github.com/conda-forge/libkml-feedstock/) on Windows but I believe it is possible to compile it there. I am not sure about `poppler`. The alternatives for a `pdf` driver are `poppler`, `PoDoFo`, and `PDFium`. In case someone wants to try to get that driver on Windows.

@conda-forge/libgdal can someone review this recipe before we merge it?

I did not create a test for te `libzma`, `pcre` and `pdf` drivers, not sure how to...

### CODE REVIEW

**Comment by kmuehlbauer** [comment]
@ocefpaf Changes seem all according to textbook.

For testing, we could think about facilitating the [gdalautotest-suite](https://trac.osgeo.org/gdal/browser/trunk/autotest). But I haven't tried this before.

**Comment by ocefpaf** [comment]
@kmuehlbauer even though I would like to run a more comprehensive test suite I am not sure our CIs can handle that. I guess that maybe only a simple `ogrinfo` test, like the one for the `kmz` would suffice.

BTW, I wanted to resolve https://github.com/conda-forge/libkml-feedstock/pull/1 before merging this, but due to the failure there I don't really want to hold this PR for much long b/c it closes https://github.com/conda-forge/gdal-feedstock/issues/120 and https://github.com/conda-forge/gdal-feedstock/issues/152

TL;DR if you agree with me above please merge this, if not I'll get back to the remaining issue next month.

**Comment by kmuehlbauer** [comment]
@ocefpaf I'll have a look into `gdalautotest` when I find time. Maybe we can run a subset of tests within a reasonable time.
